### PR TITLE
[GStreamer] Upgrade to 1.20.3

### PR DESCRIPTION
**Pull request by lucifer1004**

### CODE REVIEW

File: G/GStreamer/build_tarballs.jl [hunk 1]
```diff
@@ -42,13 +42,13 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="Glib_jll", uuid="7746bdde-850d-59dc-9ae8-88ece973131d"), v"2.59.0"; compat="2.59.0")
+    Dependency(PackageSpec(name="Glib_jll", uuid="7746bdde-850d-59dc-9ae8-88ece973131d"))
```
Comment (giordano) [inline G/GStreamer/build_tarballs.jl:45]: Please do ***not*** remove compat bounds, they're there for a good reason:  #3024

File: G/GStreamer/build_tarballs.jl [hunk 2]
```diff
@@ -42,13 +42,13 @@ products = [
...
     Dependency(PackageSpec(name="LibUnwind_jll", uuid="745a5e78-f969-53e9-954f-d19f2f74f4e3"))
     Dependency(PackageSpec(name="Elfutils_jll", uuid="ab5a07f8-06af-567f-a878-e8bb879eba5a"))
-    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"), v"6.1.2")
+    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"))
```
Comment (giordano) [inline G/GStreamer/build_tarballs.jl:48]: Same here, and this should be a compat bound: #3024

File: G/GStreamer/build_tarballs.jl [hunk 3]
```diff
@@ -42,13 +42,13 @@ products = [
...
     Dependency(PackageSpec(name="GSL_jll", uuid="1b77fbbe-d8ee-58f0-85f9-836ddc23a7a4"))
```
Comment (giordano) [inline G/GStreamer/build_tarballs.jl:49]: And also #3024

Comment (giordano) [reply G/GStreamer/build_tarballs.jl]: Why was this marked as resolved?  You didn't address this comment

**Comment by giordano** [comment]
I fixed a bunch of issues, but now the remaining problem is macOS:
```
[23:15:26] [358/507] Linking target gst/libgstreamer-1.0.0.dylib
[23:15:26] ninja: job failed: /opt/bin/x86_64-apple-darwin14-libgfortran3-cxx03/x86_64-apple-darwin14-clang  -o gst/libgstreamer-1.0.0.dylib gst/libgstreamer-1.0.0.dylib.p/meson-generated_.._gstenumtypes.c.o gst/libgstreamer-1.0.0.dylib.p/meson-generated_.._parse_lex.priv_gst_parse_yy.c.o gst/libgstreamer-1.0.0.dylib.p/meson-generated_.._parse_grammar.tab.c.o gst/libgstreamer-1.0.0.dylib.p/gst.c.o gst/libgstreamer-1.0.0.dylib.p/gstobject.c.o gst/libgstreamer-1.0.0.dylib.p/gstallocator.c.o gst/libgstreamer-1.0.0.dylib.p/gstbin.c.o gst/libgstreamer-1.0.0.dylib.p/gstbuffer.c.o gst/libgstreamer-1.0.0.dylib.p/gstbufferlist.c.o gst/libgstreamer-1.0.0.dylib.p/gstbufferpool.c.o gst/libgstreamer-1.0.0.dylib.p/gstbus.c.o gst/libgstreamer-1.0.0.dylib.p/gstcaps.c.o gst/libgstreamer-1.0.0.dylib.p/gstcapsfeatures.c.o gst/libgstreamer-1.0.0.dylib.p/gstchildproxy.c.o gst/libgstreamer-1.0.0.dylib.p/gstclock.c.o gst/libgstreamer-1.0.0.dylib.p/gstcontext.c.o gst/libgstreamer-1.0.0.dylib.p/gstcontrolbinding.c.o gst/libgstreamer-1.0.0.dylib.p/gstcontrolsource.c.o gst/libgstreamer-1.0.0.dylib.p/gstdatetime.c.o gst/libgstreamer-1.0.0.dylib.p/gstdebugutils.c.o gst/libgstreamer-1.0.0.dylib.p/gstdevice.c.o gst/libgstreamer-1.0.0.dylib.p/gstdeviceprovider.c.o gst/libgstreamer-1.0.0.dylib.p/gstdeviceproviderfactory.c.o gst/libgstreamer-1.0.0.dylib.p/gstdynamictypefactory.c.o gst/libgstreamer-1.0.0.dylib.p/gstelement.c.o gst/libgstreamer-1.0.0.dylib.p/gstelementfactory.c.o gst/libgstreamer-1.0.0.dylib.p/gsterror.c.o gst/libgstreamer-1.0.0.dylib.p/gstevent.c.o gst/libgstreamer-1.0.0.dylib.p/gstformat.c.o gst/libgstreamer-1.0.0.dylib.p/gstghostpad.c.o gst/libgstreamer-1.0.0.dylib.p/gstdevicemonitor.c.o gst/libgstreamer-1.0.0.dylib.p/gstinfo.c.o gst/libgstreamer-1.0.0.dylib.p/gstiterator.c.o gst/libgstreamer-1.0.0.dylib.p/gstatomicqueue.c.o gst/libgstreamer-1.0.0.dylib.p/gstmessage.c.o gst/libgstreamer-1.0.0.dylib.p/gstmeta.c.o gst/libgstreamer-1.0.0.dylib.p/gstmemory.c.o gst/libgstreamer-1.0.0.dylib.p/gstminiobject.c.o gst/libgstreamer-1.0.0.dylib.p/gstpad.c.o gst/libgstreamer-1.0.0.dylib.p/gstpadtemplate.c.o gst/libgstreamer-1.0.0.dylib.p/gstparamspecs.c.o gst/libgstreamer-1.0.0.dylib.p/gstpipeline.c.o gst/libgstreamer-1.0.0.dylib.p/gstplugin.c.o gst/libgstreamer-1.0.0.dylib.p/gstpluginfeature.c.o gst/libgstreamer-1.0.0.dylib.p/gstpluginloader.c.o gst/libgstreamer-1.0.0.dylib.p/gstpoll.c.o gst/libgstreamer-1.0.0.dylib.p/gstpreset.c.o gst/libgstreamer-1.0.0.dylib.p/gstprotection.c.o gst/libgstreamer-1.0.0.dylib.p/gstquark.c.o gst/libgstreamer-1.0.0.dylib.p/gstquery.c.o gst/libgstreamer-1.0.0.dylib.p/gstregistry.c.o gst/libgstreamer-1.0.0.dylib.p/gstregistrychunks.c.o gst/libgstreamer-1.0.0.dylib.p/gstpromise.c.o gst/libgstreamer-1.0.0.dylib.p/gstsample.c.o gst/libgstreamer-1.0.0.dylib.p/gstsegment.c.o gst/libgstreamer-1.0.0.dylib.p/gststreamcollection.c.o gst/libgstreamer-1.0.0.dylib.p/gststreams.c.o gst/libgstreamer-1.0.0.dylib.p/gststructure.c.o gst/libgstreamer-1.0.0.dylib.p/gstsystemclock.c.o gst/libgstreamer-1.0.0.dylib.p/gsttaglist.c.o gst/libgstreamer-1.0.0.dylib.p/gsttagsetter.c.o gst/libgstreamer-1.0.0.dylib.p/gsttask.c.o gst/libgstreamer-1.0.0.dylib.p/gsttaskpool.c.o gst/libgstreamer-1.0.0.dylib.p/gsttoc.c.o gst/libgstreamer-1.0.0.dylib.p/gsttocsetter.c.o gst/libgstreamer-1.0.0.dylib.p/gsttracer.c.o gst/libgstreamer-1.0.0.dylib.p/gsttracerfactory.c.o gst/libgstreamer-1.0.0.dylib.p/gsttracerrecord.c.o gst/libgstreamer-1.0.0.dylib.p/gsttracerutils.c.o gst/libgstreamer-1.0.0.dylib.p/gsttypefind.c.o gst/libgstreamer-1.0.0.dylib.p/gsttypefindfactory.c.o gst/libgstreamer-1.0.0.dylib.p/gsturi.c.o gst/libgstreamer-1.0.0.dylib.p/gstutils.c.o gst/libgstreamer-1.0.0.dylib.p/gstvalue.c.o gst/libgstreamer-1.0.0.dylib.p/gstparse.c.o gst/libgstreamer-1.0.0.dylib.p/gstregistrybinary.c.o -L/workspace/destdir/lib -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -shared -install_name @rpath/libgstreamer-1.0.0.dylib -compatibility_version 2004 -current_version 2004 -Wl,-rpath,/workspace/destdir/lib gst/printf/libgstprintf.a /workspace/destdir/lib/libgobject-2.0.dylib /workspace/destdir/lib/libglib-2.0.dylib /workspace/destdir/lib/libintl.dylib /workspace/destdir/lib/libgmodule-2.0.dylib -lm -ldl
[23:15:26] ninja: subcommand failed
[23:15:26] ld: warning: ignoring file gst/printf/libgstprintf.a, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x74 0x68 0x69 0x6E 0x3E 0x0A 0x2F 0x20 0x20 0x20 0x20 0x20 0x20 0x20 )
[23:15:26] Undefined symbols for architecture x86_64:
[23:15:26]   "___gst_printf_pointer_extension_set_func", referenced from:
[23:15:26]       __priv_gst_debug_init in gstinfo.c.o
[23:15:26]   "___gst_vasprintf", referenced from:
[23:15:26]       __gst_element_error_printf in gstelement.c.o
[23:15:26]       _gst_debug_log_default in gstinfo.c.o
[23:15:26]       _gst_debug_message_get in gstinfo.c.o
[23:15:26]       _gst_debug_log_get_line in gstinfo.c.o
[23:15:26]       _gst_info_vasprintf in gstinfo.c.o
[23:15:26]       _gst_info_strdup_vprintf in gstinfo.c.o
[23:15:26]       _gst_info_strdup_printf in gstinfo.c.o
[23:15:26]       ...
[23:15:26] ld: symbol(s) not found for architecture x86_64
[23:15:26] clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```
Superficially this looks like https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/398, but that was supposed to be fixed years ago, so it may not be actually related.  I don't have the time to investigate it further.

**Comment by lucifer1004** [comment]
It seems to be the `ar` issue: https://stackoverflow.com/questions/73065842/ld-warning-ignoring-file-liblua-a-building-for-macos-x86-64-but-attempting-to

**Comment by lucifer1004** [comment]
Which `ar` do we use in the OSX image?

**Comment by giordano** [comment]
Oh, I missed that. The problem is meson being broken: https://github.com/JuliaPackaging/Yggdrasil/pull/5539#issuecomment-1260165614, https://github.com/mesonbuild/meson/issues/10823#issuecomment-1260194171

File: G/GStreamer/build_tarballs.jl [hunk 4]
```diff
@@ -42,13 +49,15 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="Glib_jll", uuid="7746bdde-850d-59dc-9ae8-88ece973131d"), v"2.59.0"; compat="2.59.0")
+    # Need a host gettext for msgfmt
+    HostBuildDependency("Gettext_jll")
+    Dependency(PackageSpec(name="Glib_jll", uuid="7746bdde-850d-59dc-9ae8-88ece973131d"); compat="2.68.1")
     Dependency(PackageSpec(name="LibUnwind_jll", uuid="745a5e78-f969-53e9-954f-d19f2f74f4e3"))
     Dependency(PackageSpec(name="Elfutils_jll", uuid="ab5a07f8-06af-567f-a878-e8bb879eba5a"))
```
Comment (giordano) [inline G/GStreamer/build_tarballs.jl:56]: Elfutils is only for Linux. I'm not even sure it's actually used?

Comment (lucifer1004) [reply G/GStreamer/build_tarballs.jl:56]: I don't know either. I just kept it as it was.

Comment (giordano) [reply G/GStreamer/build_tarballs.jl:62]:
```suggestion
    Dependency(PackageSpec(name="Elfutils_jll", uuid="ab5a07f8-06af-567f-a878-e8bb879eba5a"); platforms=filter(Sys.islinux, platforms))
```

File: G/GStreamer/build_tarballs.jl [hunk 5]
```diff
@@ -42,13 +55,15 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="Glib_jll", uuid="7746bdde-850d-59dc-9ae8-88ece973131d"), v"2.59.0"; compat="2.59.0")
+    # Need a host gettext for msgfmt
+    HostBuildDependency("Gettext_jll")
+    Dependency(PackageSpec(name="Glib_jll", uuid="7746bdde-850d-59dc-9ae8-88ece973131d"); compat="2.68.1")
     Dependency(PackageSpec(name="LibUnwind_jll", uuid="745a5e78-f969-53e9-954f-d19f2f74f4e3"))
```
Comment (giordano) [inline G/GStreamer/build_tarballs.jl:61]:
```suggestion
    Dependency(PackageSpec(name="LibUnwind_jll", uuid="745a5e78-f969-53e9-954f-d19f2f74f4e3"); platforms=filter(p -> Sys.islinux(p) || Sys.isfreebsd(p), platforms))
```

**Comment by lucifer1004** [comment]
@giordano Seems all issues have been fixed.

**Pull request merged**